gatsby-node.js

```diff
@@ -51,8 +53,82 @@ const makePage = ({ chamber, pageData, createPage, legislatorId }) => {
 }
 
 // create individual legislator pages
-exports.createPages = async function ({ actions: { createPage } }) {
-  ;[
+exports.createPages = async function ({ actions, graphql }) {
+  const { createPage } = actions
+
+  //TODO - other sessions
+  let sessionYear = 2023
+  // sponsorships
+  const getLegislatorById = (id) => {
+    return (
+      houseLegislators.find((legislator) => legislator.id === id) ??
+      senateLegislators?.find((legislator) => legislator.id === id)
+    )
+  }
+
+  let sponsoredBills = Object.entries(legislationData[sessionYear].sponsoredBills)
+
+  const allSponsoredBillsTemplate = path.resolve(`./src/components/sponsorships/index.js`)
+
+  createPage({
+    path: `/sponsorships/all-bills`,
+    component: allSponsoredBillsTemplate,
+    context: {
+      sponsoredBills,
+    },
+  })
+
+  const sponsoredBillTemplate = path.resolve(`./src/components/sponsorships/sponsorships.js`)
+
+  let sponsors = Object.entries(legislationData[sessionYear].sponsorship).map((sponsorshipItem) => {
+    const [id, sponsorshipData] = sponsorshipItem
+    let legislatorData = getLegislatorById(sponsorshipData.id)
+    return {
+      sponsorshipData: { ...sponsorshipData.data, ...sponsorshipData.score },
+      ...legislatorData,
+    }
+  })
+
+  sponsoredBills.forEach((sponsoredBill) => {
+    let sortedSponsors = []
+    const [billNumber, billData] = sponsoredBill
+    let otherBillNames = ''
+    sortedSponsors = sponsors
+      .filter((sponsor) => {
+        if (!sponsor.id) return false
+        for (const bills in sponsor.sponsorshipData) {
+          if (bills.includes(billNumber) && sponsor.sponsorshipData[bills]) {
+            otherBillNames = bills
```

Comment:
Can you explain what `otherBillNames` is/does?

Reply:
here I am filtering out bills that are the same just house or senate versions so theyre not displayed twice, but still keeping track of their other name (H---/S---) so we can link to them. It seems like we might have a way to account for this but I wasnt sure